homeassistant/components/kermi/__init__.py

```diff
@@ -0,0 +1,107 @@
+"""Kermi heatpump integration for Home Assistant."""
+
+from datetime import timedelta
+from functools import partial
+import logging
+
+from pymodbus.client import AsyncModbusTcpClient
+from pymodbus.exceptions import ConnectionException, ModbusIOException
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, MODBUS_REGISTERS
+
+# Create a logger
+logger = logging.getLogger(__name__)
+
+PLATFORMS = ["water_heater"]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up kermi from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+
+    # Create a ModbusTcpClient
+    client = AsyncModbusTcpClient(host=entry.data[CONF_HOST], port=502)
```

Comment:
you seem to only use the client in the coordinator, so I'd init it in there and don't store in the config entry again.